### ISSUE TITLE
Fail to train on DVSGesture

### ISSUE THREAD

**Issue by haohq19** [AIRLab-POLIMI/farse-cnn#4]
Dear authors,

Thank you for your wonderful paper! I'm very interested in work and is trying to train on DVSGesture. However, there is an error. I tried to fix this by changing line 18-28 in utils/data_utils.py from:
```
elif dataset_name == 'DVSGesture100ms':
        # Event format: <xypt>
        dvsg_train = tonic.datasets.DVSGesture(save_to='data', train=True)
        dvsg_test = tonic.datasets.DVSGesture(save_to='data', train=False)
        if val_split:
            train_dataset = DatasetWindows(dvsg_train, split_path="data/dvsgesture-split-train-w01-s005-subset08.json", transform=train_transform)
            val_dataset = DatasetWindows(dvsg_train, split_path="data/dvsgesture-split-train-w01-s005-subset02.json",  transform=val_transform)
        else:
            train_dataset = DatasetWindows(dvsg_train, split_path="data/dvsgesture-split-train-w01-s005.json",  transform=train_transform)
            val_dataset = None
        test_dataset = DatasetWindows(dvsg_test, split_path="data/dvsgesture-split-test-w01-s005.json",  transform=test_transform)
```
to
```
elif dataset_name == 'DVSGesture100ms':
        # Event format: <xypt>
        dvsg_train = tonic.datasets.DVSGesture(save_to='data', train=True)
        dvsg_test = tonic.datasets.DVSGesture(save_to='data', train=False)
        if val_split:
            train_dataset = DatasetWindows(dvsg_train, split_path="data/dvsgesture-split-train-w01-s005-subset08.json", split_windows_size=0.1, split_windows_stride=0.1, transform=train_transform)
            val_dataset = DatasetWindows(dvsg_train, split_path="data/dvsgesture-split-train-w01-s005-subset02.json", split_windows_size=0.1, split_windows_stride=0.1, transform=val_transform)
        else:
            train_dataset = DatasetWindows(dvsg_train, split_path="data/dvsgesture-split-train-w01-s005.json", split_windows_size=0.1, split_windows_stride=0.1, transform=train_transform)
            val_dataset = None
        test_dataset = DatasetWindows(dvsg_test, split_path="data/dvsgesture-split-test-w01-s005.json", split_windows_size=0.1, split_windows_stride=0.1, transform=test_transform)
```

However, I could still not get the results on DVSGesture. (For me, the accuracy is 0.88). I want to kindly ask if there is something I'm wrong. Thank you very much for your work!

**Comment by rcrsnt** [comment]
Hello, thank you for using our code.
You are correct that there is an error there. The DatasetWindows class will automatically generate a .json file with the metadata used for splitting samples, but the first time that it is run, we should provide the window size and stride. 
If you want to replicate our experiments exactly, you should use `split_windows_size=0.1, split_windows_stride=0.05`.

Also, please notice that for DVSGesture we train on these windows, but then test on the full sequences, which can be replicated by setting `dataset_name: 'DVSGesture'` in your test cfg. The testing script will report the mode accuracy, which is computed by taking the most common prediction given by the model over the entire sample length.

**Comment by haohq19** [comment]
Thank you for your kind reply. Now I can reproduce the results on DVSGesture. Thank you for your wonderful paper!
